pmd-core/src/main/java/net/sourceforge/pmd/lang/ast/AntlrNode.java

```diff
@@ -0,0 +1,153 @@
+/**
+ * BSD-style license; for more info see http://pmd.sourceforge.net/license.html
+ */
+
+package net.sourceforge.pmd.lang.ast;
+
+import java.util.List;
+
+import org.jaxen.JaxenException;
+
+/**
+ * Base interface for all Antlr-based implementation of Node interface.
+ * <p>
+ * Initially all the methods implemented here will be no-op due to scope limitations
+ */
+public interface AntlrNode extends Node {
+
+    @Override
+    default void jjtOpen() {
+        throw new UnsupportedOperationException("Won't be needed on Antlr implementation");
+    }
+
+    @Override
+    default void jjtClose() {
+        throw new UnsupportedOperationException("Won't be needed on Antlr implementation");
+    }
+
+    @Override
+    default void jjtSetParent(final Node parent) {
+        throw new UnsupportedOperationException("Out of scope for antlr current implementations");
+    }
+
+    @Override
+    default Node jjtGetParent() {
+        return null; // TODO: review this
+    }
+
+    @Override
+    default void jjtAddChild(final Node child, final int index) {
+        throw new UnsupportedOperationException("Out of scope for antlr current implementations");
+    }
+
+    @Override
+    default void jjtSetChildIndex(final int index) {
+        throw new UnsupportedOperationException("Out of scope for antlr current implementations");
+    }
+
+    @Override
+    default int jjtGetChildIndex() {
+        throw new UnsupportedOperationException("Out of scope for antlr current implementations");
+    }
+
+    @Override
+    default Node jjtGetChild(final int index) {
+        throw new UnsupportedOperationException("Out of scope for antlr current implementations");
+    }
+
+    @Override
+    default int jjtGetNumChildren() {
+        throw new UnsupportedOperationException("Out of scope for antlr current implementations");
+    }
+
+    @Override
+    default int jjtGetId() {
```

Comment:
This is very JJTree specific too. The id only serves as an index in the node name lookup for JJTree

Reply:
Nice, we should deprecate it along the other only-javaCC methods, i'll send another PR doing this into 6.X, thank you for the information 😄